src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,207 @@
+package duke.storage;
+
+import duke.task.*;
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+
+
+/**
+ * This class handles the storage and retrieval of tasks to and from a file.
+ */
+public class Storage {
+
+    private String filePath;
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filePath The path of the file to be used for storage.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the file and returns a TaskArray containing the parsed tasks.
+     *
+     * @return A TaskArray containing the loaded tasks.
+     */
+    public TaskArray load() {
+        // Create the folder/file if it doesn't exist
+        boolean createdFolder = createFolder(this.filePath);
+        boolean createdFile =createFile(this.filePath);
+
+        ArrayList<String> result = new ArrayList();
+        ArrayList<String> tobeProcessedArray = scanFile(this.filePath);
+
+
+
+        return parseData(tobeProcessedArray);
+
+    }
+
+    /**
+     * Creates a new file at the specified file path if it doesn't exist.
+     *
+     * @param filePath The path of the file to be created.
+     * @return True if the file already exists, false if a new file was created.
+     */
+    public boolean createFile(String filePath) {
+        File file = new File(filePath);
+        if (!file.exists()) {
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
```

Comment:
Not too sure whether this format would be allowed, as the Coding Standards state it should be done in another method instead.

![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/94635140/e6f2537b-09cd-469d-bc04-48016e4807d3)

It is right above the part on the white space issue.